src/mds/MDCache.cc

```diff
@@ -10120,6 +10120,7 @@ void MDCache::scan_stray_dir(dirfrag_t next)
 	}
       }
     }
+    next.frag = frag_t();
```

Comment:
Looks correct. You found this bug as not all stray inodes were getting evaluated, yes?

Reply:
Yes, I did a test that would produce a lot of stray dentries and found out that many of them didn't get evaluated after MDS restarted.